docs/framework/dimensions.rst

```diff
@@ -67,23 +67,40 @@ dimension, than we will end up with just a dimensionless quantity:
 Quantity Kinds
 ++++++++++++++
 
-Quantity kinds behave the same as quantities for addition and subtraction.
-The same behavior is also provided for multiplication with, and division by
-a :term:`scalable number`.
+Quantity kinds behave the same as quantities for all operations,
+except that the quantity types in the operators' declarations
+are quantity kind types instead.
+Additionally, for the dimensional analysis operators,
+you can use a quantity argument instead of a quantity kind.
 
-Multiplication and division with a quantity (but not a quantity kind) is allowed.
-The result is a quantity kind with the appropriate dimension
-and related to the original quantity kind:
+.. code-block::
+    :emphasize-lines: 8-9
+
+    struct height_kind : kind<height_kind, dim_length> {};
+    struct rate_of_climb_kind : derived_kind<rate_of_climb_kind, height_kind, dim_speed> {};
+
+    template <Unit U, QuantityValue Rep = double> using height = quantity_kind<height_kind, U, Rep>;
+    template <Unit U, QuantityValue Rep = double> using rate_of_climb = quantity_kind<rate_of_climb_kind, U, Rep>;
+
+    height h{100 * m};
+    rate_of_climb rate = h / (25 * s);
+      // quantity_kind<rate_of_climb_kind, si::metre_per_second, int>(4 * m / s)
 
 .. code-block::
-    :emphasize-lines: 5-6
+    :emphasize-lines: 8-12
+
+    struct width_kind : kind<width_kind, dim_length> {};
+    struct horizontal_area_kind : derived_kind<horizontal_area_kind, width_kind, dim_area> {};
```

Comment:
BTW, it would be good to have some docs on `derived_kind` if we decide to keep it.

Reply:
BTW2, do you think we should `deduced_unit` to a `derived_unit`? It might be a better name?

Reply:
> I am not sure if I understand why we need a `derived_kind` and why `kind` takes a `Dimension` in the first place? Can't a `kind` be just a simple tag type?

I think it mostly is a tag type. It just happens to wrap a `Dimension` and provide convenient nested aliases. There's also the downcasting facility applied to it. If it didn't wrap a `Dimension`, we'd end up with longer specialization names, and the lack of symmetry would make https://github.com/mpusz/units/discussions/213 impossible.

Reply:
> BTW, it would be good to have some docs on `derived_kind` if we decide to keep it.

`derived_kind` establishes a relation, in code, between kinds. Coupled with the downcasting facility, that's how we're able to have the operators result in the related quantity kinds.

Reply:
> If it didn't wrap a Dimension, we'd end up with longer specialization names

I am not sure about the above. If the kind is just a tag type (independent from the dimension) then there is no need for a downcasting facility, right?

Reply:
Not just a tag type, just "mostly". We'd still need a way to establish the relationship between kinds. I chose the downcasting facility.